util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/assertion/SearchHitsAssert.java

```diff
@@ -103,6 +103,12 @@ public SearchHitsAssert<H> hasDocRefHitsAnyOrder(String typeName, String firstId
 		} );
 	}
 
+	public SearchHitsAssert<H> containsListHitsAnyOrder(String typeName, String firstId, String... otherIds) {
+		return containsListHitsAnyOrder( ctx -> {
+			ctx.doc( typeName, firstId, otherIds );
+		} );
+	}
+
```

Comment:
Let's avoid these "incomplete" checks where extra, unwanted matches could go unnoticed... Could you remove these `contains*` methods you added, and use something like this in your checks instead? This allows checking for documents from multiple indexes in one assertion.

https://github.com/hibernate/hibernate-search/blob/6199847ddb9a404df03995d244bff758a18dce56/integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/AbstractPredicateTypeCheckingAndConversionIT.java#L232-L235